wasi/src/main/java/com/dylibso/chicory/wasi/WasiPreview1.java

```diff
@@ -194,12 +197,8 @@ public int clockTimeGet(Memory memory, int clockId, long precision, int resultPt
         logger.tracef("clock_time_get: [%s, %s, %s]", clockId, precision, resultPtr);
         switch (clockId) {
             case WasiClockId.REALTIME:
-                Instant now = clock.instant();
-                long epochNanos = SECONDS.toNanos(now.getEpochSecond()) + now.getNano();
-                memory.writeLong(resultPtr, epochNanos);
-                return wasiResult(WasiErrno.ESUCCESS);
             case WasiClockId.MONOTONIC:
-                memory.writeLong(resultPtr, System.nanoTime());
+                memory.writeLong(resultPtr, clockTime(clockId));
```

Comment:
Would it be possible to use `clockTime`(or something similar) everywhere instead of `System.nanotime()`?

Reply:
This method is a helper method that is shared with `clock_time_get()` to get the realtime or monotonic clock. The other usages of `System.nanoTime()` are in `poll_oneoff()` to measure how long we have been sleeping. As I mentioned in the other comment, mocking that seems insufficient as we directly call `Thread.sleep()`. I'd need to think more about how those interact.